lib/internal/errors.js

```diff
@@ -492,18 +523,33 @@ function E(sym, val, def, ...otherClasses) {
   // Special case for SystemError that formats the error message differently
   // The SystemErrors only have SystemError as their base classes.
   messages.set(sym, val);
-  if (def === SystemError) {
-    def = makeSystemErrorWithCode(sym);
-  } else {
-    def = makeNodeErrorWithCode(def, sym);
-  }
+
+  const ErrClass = def === SystemError ?
+    makeSystemErrorWithCode(sym) :
+    makeNodeErrorWithCode(def, sym);
 
   if (otherClasses.length !== 0) {
-    otherClasses.forEach((clazz) => {
-      def[clazz.name] = makeNodeErrorWithCode(clazz, sym);
-    });
+    if (otherClasses.includes(HideStackFramesError)) {
```

Comment:
@mcollina
@anonrig

Should we use primordials?

Reply:
I would recommend we do.